app_dart/lib/src/service/datastore.dart

```diff
@@ -247,15 +247,10 @@ class DatastoreService {
   Future<void> insert(List<Model<dynamic>> rows) async {
     final List<List<Model<dynamic>>> shards = await shard(rows);
     for (List<Model<dynamic>> shard in shards) {
-      await runTransactionWithRetries(
```

Comment:
Can we keep the retry and remove the `transaction.commit()` only?

Reply:
I suspect that we might be getting a commit from somewhere else or that Datastore is committing these transactions. I have not found for sure why this specific commit is causing the issue but that intuitively makes sense. 

Now with the retries I am not sure this makes any difference since if we look at the code we retryIf on AbortedTransactionError which we see and which means since are not wrapping the full call we are just issuing the commit every time. I would rather track it down first than just add the retry back in at this point.

Reply:
Huh, I am revisiting why we added the transaction.commit() logic, which was intended to fix the similar/same issue: https://github.com/flutter/flutter/issues/121413 `The transaction was aborted.`

Let's land it here as it mitigates our issues now. But something sounds we need to follow up with Datastore team to find out the culprit.